ui/src/layouts/storage/send_files_components.rs

```diff
@@ -74,14 +73,21 @@ pub fn send_files_from_chat_topbar(
     None
 }
 
-pub fn add_remove_file_to_send(files_selected_to_send: UseRef<Vec<String>>, file_path: String) {
-    let mut files_selected = files_selected_to_send.write();
-    if let Some(index) = files_selected
-        .iter()
-        .position(|path| path.clone() == file_path.clone())
-    {
-        files_selected.remove(index);
-    } else if files_selected.len() < MAX_FILES_PER_MESSAGE {
-        files_selected.push(file_path.clone());
+pub fn add_remove_file_to_send(storage_controller: UseRef<StorageController>, file_path: String) {
```

Comment:
`add_remove` is the same as `toggle`. `toggle_selected_file` is a better name. 

Also, you should use references for the function arguments if you can. `storage_controller` can be a `&mut` and `file_path` can be a `&str`.